pep-0604.rst

```diff
@@ -73,13 +74,59 @@ Here are some examples of what we can do with this feature.
     assert str | int == Union[str,int]
     assert str | int | float == Union[str, int, float]
 
-    assert isinstance("", int | str)
-    assert issubclass(bool, int | float)
+    # Instead of
+    # typing.List[typing.Union[str, int]]
+    typing.List[str | int]
+
+    # list[typing.Union[str, int]] (Currently TypeError: 'type' object has no attribute '__getitem__')
+    list[int | str]
+
+    # Instead of typing.Dict[str, typing.Union[int, float]]
+    typing.Dict[str, int | float]
+    dict[str, int | float]
+
+The new syntax should be accepted for calls to `isinstance` as long as the Union items are valid arguments to `isinstance` themselves.
+
+isinstance::
+  # valid
+  isinstance("", int | str)
+
+
+  # invalid
+  isinstance(2, list[int]) # TypeError: isinstance() argument 2 cannot be a parameterized generic
+  isinstance(1, int | list[int])
+
+subclass::
+  # valid
+  assert issubclass(bool, int | float)
+
+  # invalid
+
+The existing `typing.Union` and `|` syntax should be equal.::
+  int | str == typing.Union[int, str]
+
+  typing.Union[int, int] == int
+  int | int == int
+
+The order of the items in the Union should not matter for equality.::
+  (int | str) == (str | int)
+  (int | str | float) == typing.Union[str, float, int]
+
+
+Optional values should be equal to the new union syntax, for example: ::
+  None | t == typing.Optional[t]
+
+A new Union.__str__() method should be implemented.::
+  (int | list[str]).__str__()
```

Comment:
I would spell this using either `str(...)` or `print(...)`. But actually, it should be a new `__repr__` -- by defining just `__repr__` both `str()` and `repr()` will produce the same output. (And the REPL uses `repr()`.)